common/src/main/java/com/scottlogic/deg/common/profile/fields/Fields.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2019 Scott Logic Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.scottlogic.deg.common.profile.fields;
+
+import com.scottlogic.deg.common.ValidationException;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Fields implements Iterable<Field>
+{
+    private final Map<String, Field> fieldMap;
+
+    private Fields(Map<String, Field> fieldMap)
+    {
+        this.fieldMap = fieldMap;
+    }
+
+    public static Fields create(Collection<Field> fields)
```

Comment:
What is the incentive behind using a static `create(Collection<Field>)` method instead of putting this in the constructor?